### PR TITLE
Add `Jetpack_Color` to the `classic-theme-helper` package

### PR DESCRIPTION
**Pull request by monsieur-z**
Fixes https://github.com/Automattic/vulcan/issues/398

## Proposed changes:

The `Jetpack_Color` class will no longer be present in the Jetpack plugin but needs to stay available to simple sites, WoA sites and self-hosted sites with specific themes. This PR accounts for this by copying the class to the `classic-theme-helper` package.

In more detail, the PR:
- Copies  `class.color.php` to the `classic-theme-helper` package
- Fixes and ignores Phan warnings
- Add an existence guard to `Jetpack_Color` and `Tonesque`
- Loads `Jetpack_Color`
- Loads `Tonesque` if the theme supports it

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

Project thread: pfwV0U-e1-p2

## Does this pull request change what data or activity we track or use?

n/a

## Testing instructions:

Simple sites load the `classic-theme-helper` package. Let's test that a simple site with a theme that uses `Jetpack_Color`, such as Ryu, loads the class from the package when we simulate Jetpack not being installed.

### Prerequisites
- Spin up a simple site that points to your sandbox
- Install the Ryu theme
- Make sure you can see your sandbox logs
- Don't apply this PR just yet

### Prevent Jetpack from loading `Jetpack_Color`
- I don't know if there's such a thing as disabling Jetpack on simple sites. If there is, do it.
- Otherwise, let's disable the loading of `Jetpack_Color`:
  - Open the file `wp-content/mu-plugins/jetpack-plugin/production/modules/theme-tools.php` in your sandbox
  - In the `jetpack_load_theme_tools` function, comment out the loading of `tonesque.php`
- At this point, `Jetpack_Color` should not be required anywhere. Let's check that:
  - In your sandbox, open the file `wp-content/themes/pub/ryu/functions.php`
  - Update the function below so we can log values

```
function ryu_load_bundled_tonesque() {
  error_log( class_exists('Jetpack_Color') );
  error_log( class_exists('Tonesque') ); 
  
  if ( ! class_exists( 'Tonesque' ) ) {
    require( get_template_directory() . '/inc/tonesque.php' );
  }
}
```

- Visit the frontend of your site
- You should see in the logs that `class_exists` returns false in both instances

### Load `Jetpack_Color` from the package
- Apply this PR to your sandbox
  - Checkout this branch locally and sync it with your sandbox by running `jetpack rsync mu-wpcom-plugin`
  - Or use the `jetpack-downloader` command directly in your sandbox
- Refresh the frontend of your site
- You should see in the logs that `class_exists` returns true in both instances

### CODE REVIEW

File: projects/packages/classic-theme-helper/src/class-main.php [hunk 1]
```diff
@@ -39,7 +40,8 @@ public static function init() {
 			self::$instance = new Main();
 			self::$instance->load_modules();
 			// TODO Commenting below since we still load them from theme-tools module
-			// add_action( 'init', array( __CLASS__, 'jetpack_load_theme_tools' ), 30 );
+			add_action( 'init', array( __CLASS__, 'jetpack_load_theme_tools' ), 30 );
```
Comment (monsieur-z) [inline projects/packages/classic-theme-helper/src/class-main.php:43]: The `Jetpack_Color` and `Tonesque` classes loaded from the theme-tools module have existence guards so this is  safe.

File: projects/plugins/wpcomsh/.phan/config.php [hunk 2]
```diff
@@ -14,6 +14,9 @@
 	dirname( __DIR__ ),
 	array(
 		'exclude_file_regex'    => array( 'tests/lib/mocks' ),
+		'exclude_file_list'     => array(
```
Comment (monsieur-z) [inline projects/plugins/wpcomsh/.phan/config.php:17]: Prevent the Phan warning `PhanRedefinedClassReference` caused by the `Jetpack_Color` being currently defined twice.

File: projects/packages/classic-theme-helper/_inc/lib/class.color.php [hunk 3]
```diff
@@ -0,0 +1,911 @@
...
+ * Represents a color value, and converts between RGB/HSV/XYZ/Lab/HSL
+ *
+ * Example:
+ * $color = new Jetpack_Color(0xFFFFFF);
+ *
+ * @author Harold Asbridge <hasbridge@gmail.com>
+ * @author Matt Wiebe <wiebe@automattic.com>
+ * @license https://www.opensource.org/licenses/MIT
+ *
+ * @package automattic/jetpack
```
Comment (coder-karen) [inline projects/packages/classic-theme-helper/_inc/lib/class.color.php:14]: I believe this should be the Classic Theme Helper package here. I notice there are a few other files that include Jetpack as the package, possibly all from the same file move in the past.
```suggestion
 * @package automattic/jetpack-classic-theme-helper
```

File: projects/packages/classic-theme-helper/src/class-main.php [hunk 4]
```diff
@@ -39,7 +40,8 @@ public static function init() {
 			self::$instance = new Main();
 			self::$instance->load_modules();
 			// TODO Commenting below since we still load them from theme-tools module
-			// add_action( 'init', array( __CLASS__, 'jetpack_load_theme_tools' ), 30 );
```
Comment (coder-karen) [inline projects/packages/classic-theme-helper/src/class-main.php:42]: Might be worth moving the TODO comment further down now so it looks like it is just for the remaining commented out line?

**Review by coder-karen — APPROVED** [review]
This works well for me. 

I did notice that following the testing instructions commenting out the loading of Tonesque for some reason resulting in false for class Color but true for Tonesque (class exists) for me.

What I did instead - commented out the code in the tonesque file in` jetpack-plugin/production/_inc/lib/tonesque.php`, as well as in `class.color.php`
Also commented out the code in `class.color.php`.

If I load the changes only from the mu-wpcom plugin, both classes return true.

If I load the new changes only from the Jetpack plugin, I get false for Color and true for Tonesque - but I get that without any changes applied either (default behaviour). So this looks ok to me.

I attempted to test on Atomic too, though I couldn't edit the Ryu theme files. But I used the Color Posts plugin which uses Jetpack Color and noticed no issues regardless of whether I just updated Jetpack or Jetpack Mu WPcom, or both.

Added a couple of comments but they're not blockers.

**Pull request merged**